### PR TITLE
Add support for tty. Fix path_open

### PR DESCRIPTION
**Pull request by syrusakbary**
* [x] Add support for tty.
* [x] Fix `path_open`
* [x] Implement `fd_seek`
* [x] Fix `setBigUint64` polyfill (for Safari)
* [x] Add `getBigUint64` polyfill implementation

This is needed for running `openssl` in the browsers

### CODE REVIEW

File: packages/wasi/lib/wasi.ts [hunk 1]
```diff
@@ -421,6 +422,7 @@ class WASI {
 
     const CHECK_FD = (fd: number, rights: BigIntPolyfillType) => {
       const stats = stat(this, fd);
+      // console.log(`CHECK_FD: stats.real: ${stats.real}, stats.path:`, stats.path);
```
Comment (torch2424) [inline packages/wasi/lib/wasi.ts]: Let's remove commented console logs 😄

File: packages/wasi/lib/wasi.ts [hunk 2]
```diff
@@ -1318,6 +1319,28 @@ class WASI {
         return WASI_ENOSYS;
       }
     };
+    // Wrap each of the imports with a recoverable WASI error
+    Object.keys(this.wasiImport).forEach((key: string) => {
+      const prevImport = this.wasiImport[key];
+      this.wasiImport[key] = function(...args: any[]) {
+        // console.log(`wasiImport called: ${key} (${args})`);
```
Comment (torch2424) [inline packages/wasi/lib/wasi.ts]: Same here

File: packages/wasi/lib/wasi.ts [hunk 3]
```diff
@@ -1318,6 +1319,28 @@ class WASI {
...
+        try {
+          let result = prevImport(...args);
+          // console.log(` => ${result}`);
```
Comment (torch2424) [inline packages/wasi/lib/wasi.ts]: Same here

File: packages/wasi/lib/wasi.ts [hunk 4]
```diff
@@ -1318,6 +1319,28 @@ class WASI {
         return WASI_ENOSYS;
       }
     };
+    // Wrap each of the imports with a recoverable WASI error
+    Object.keys(this.wasiImport).forEach((key: string) => {
```
Comment (torch2424) [inline packages/wasi/lib/wasi.ts]: Nice solution for this 😄

File: packages/wasmfs/lib/index.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 // An alternative fs for the browser and testing
 import { createFsFromVolume, IFs } from "memfs";
 import { Volume } from "memfs/lib/volume";
+import "./node_sync_emit";
```
Comment (torch2424) [inline packages/wasmfs/lib/index.ts]: I see this file is imported, but not committed? 🤔

File: packages/wasm-terminal/lib/command/wasi-command.ts [hunk 6]
```diff
@@ -74,9 +73,12 @@ export default class WASICommand extends Command {
 
     // Bind our stdinRead / stdoutWrite
     this.wasmFs = wasmFs;
-    this.wasmFs.volume.fds[0].read = this.stdinRead.bind(this);
-    this.wasmFs.volume.fds[1].write = this.stdoutWrite.bind(this);
-    this.wasmFs.volume.fds[2].write = this.stdoutWrite.bind(this);
+    this.wasmFs.volume.fds[0].node.read = this.stdinRead.bind(this);
+    this.wasmFs.volume.fds[1].node.write = this.stdoutWrite.bind(this);
+    this.wasmFs.volume.fds[2].node.write = this.stdoutWrite.bind(this);
+    const ttyFd = this.wasmFs.volume.openSync("/dev/tty", "w+");
```
Comment (torch2424) [inline packages/wasm-terminal/lib/command/wasi-command.ts]: Oh huh, interesting you had to do this. Thank you for solving that! 😄

File: packages/wasi/lib/wasi.ts (hunk 1 above)
Comment (syrusakbary) [reply packages/wasi/lib/wasi.ts]: This is incredibly useful for me each time I debug.

I removed it before, but each time I try to fix something I end up rewriting it again and again.
Do you feel ok with keeping it for now and removing once WASI is fully working?

File: packages/wasmfs/lib/index.ts (hunk 5 above)
Comment (syrusakbary) [reply packages/wasmfs/lib/index.ts]: It's already present on the main source (I just import it to make sure `emit` is synchronous).

File: packages/wasi/lib/wasi.ts (hunk 1 above)
Comment (torch2424) [reply packages/wasi/lib/wasi.ts]: @syrusakbary Sure! 😄

File: packages/wasmfs/lib/index.ts (hunk 5 above)
Comment (torch2424) [reply packages/wasmfs/lib/index.ts]: Sounds good 😄

**Comment by torch2424** [comment]
@syrusakbary Let me know when this is good to go, I can do a final review and merge 😄

**Pull request merged**